platforms/enterprise/enterprise/src/test/groovy/org/gradle/internal/enterprise/DevelocityBuildLifecycleServiceTest.groovy

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.enterprise
+
+import org.gradle.api.configuration.BuildFeatures
+import org.gradle.api.internal.configuration.DefaultBuildFeature
+import org.gradle.api.invocation.Gradle
+import org.gradle.api.invocation.GradleLifecycle
+import org.gradle.internal.enterprise.impl.DefaultDevelocityBuildLifecycleService
+import org.gradle.util.TestUtil
+import spock.lang.Specification
+
+class DevelocityBuildLifecycleServiceTest extends Specification {
```

Comment:
🤔  While it's okay to have a unit test for this, we really should have an e2e test as well. We don't have to do it in this PR, but then we should really do it in a follow up.

The use case for the e2e test is to use the DV callback with non-serializable state, which should work in vintage and CC, and should blow up with IP.

We already have some tests for this callback:
https://github.com/gradle/gradle/blob/ac2a744bfa14510f127071494ad4bb48b81cf828/platforms/enterprise/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/DevelocityBuildLifecycleServiceIntegrationTest.groovy#L23